tests/phpunit/elementor/modules/atomic-widgets/prop-type-migrations/test-migration-interpreter.php

```diff
@@ -0,0 +1,807 @@
+		$migration = [
+			'up' => [
+				[
+					'op' => [ 'fn' => 'set', 'path' => 'settings.valid', 'value' => true ],
+					'condition' => [ 'fn' => 'is_object', 'path' => 'settings.value' ],
+				],
+			],
+		];
+
+		// Act
+		$result = Migration_Interpreter::run( $migration, $data );
+
+		// Assert
+		$this->assertTrue( $result['settings']['valid'] );
+	}
+
+	public function test_condition_is_array() {
+		// Arrange
+		$data = [ 'settings' => [ 'items' => [ 'a', 'b', 'c' ] ] ];
+		$migration = [
+			'up' => [
+				[
+					'op' => [ 'fn' => 'set', 'path' => 'settings.valid', 'value' => true ],
+					'condition' => [ 'fn' => 'is_array', 'path' => 'settings.items' ],
+				],
+			],
+		];
+
+		// Act
+		$result = Migration_Interpreter::run( $migration, $data );
+
+		// Assert
+		$this->assertTrue( $result['settings']['valid'] );
+	}
+
+	public function test_condition_and() {
+		// Arrange
+		$data = [ 'settings' => [ '$$type' => 'size', 'value' => 150 ] ];
+		$migration = [
+			'up' => [
+				[
+					'op' => [ 'fn' => 'set', 'path' => 'settings.processed', 'value' => true ],
+					'condition' => [
+						'fn' => 'and',
+						'conditions' => [
+							[ 'fn' => 'equals', 'path' => 'settings.$$type', 'value' => 'size' ],
+							[ 'fn' => 'is_primitive', 'path' => 'settings.value' ],
+						],
+					],
+				],
+			],
+		];
+
+		// Act
+		$result = Migration_Interpreter::run( $migration, $data );
+
+		// Assert
+		$this->assertTrue( $result['settings']['processed'] );
+	}
+
+	public function test_condition_or() {
+		// Arrange
+		$data = [ 'settings' => [ '$$type' => 'html' ] ];
+		$migration = [
+			'up' => [
+				[
+					'op' => [ 'fn' => 'set', 'path' => 'settings.isText', 'value' => true ],
+					'condition' => [
+						'fn' => 'or',
+						'conditions' => [
+							[ 'fn' => 'equals', 'path' => 'settings.$$type', 'value' => 'string' ],
+							[ 'fn' => 'equals', 'path' => 'settings.$$type', 'value' => 'html' ],
+						],
+					],
+				],
+			],
+		];
+
+		// Act
+		$result = Migration_Interpreter::run( $migration, $data );
+
+		// Assert
+		$this->assertTrue( $result['settings']['isText'] );
+	}
+
+	public function test_condition_with_wildcard() {
+		// Arrange
+		$data = [
+			'settings' => [
+				'link' => [ '$$type' => 'link' ],
+				'button' => [ '$$type' => 'link' ],
+				'text' => [ '$$type' => 'string' ],
+			],
+		];
+		$migration = [
+			'up' => [
+				[
+					'op' => [ 'fn' => 'set', 'path' => 'settings.*.processed', 'value' => true ],
+					'condition' => [ 'fn' => 'equals', 'path' => 'settings.*.$$type', 'value' => 'link' ],
+				],
+			],
+		];
+
+		// Act
+		$result = Migration_Interpreter::run( $migration, $data );
+
+		// Assert
+		$this->assertTrue( $result['settings']['link']['processed'] );
+		$this->assertTrue( $result['settings']['button']['processed'] );
+		$this->assertArrayNotHasKey( 'processed', $result['settings']['text'] );
+	}
+
+	public function test_interpret_down_direction() {
```

Comment:
IS this the only test that checks the down direction?
Do we want to increase the downgrade test coverage?